appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java

```diff
@@ -43,7 +47,32 @@ public NewYoungAndroidProjectWizard() {
     // Initialize the UI
     setStylePrimaryName("ode-DialogBox");
 
-    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel());
+    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
+      @Override
+      public boolean validate(String value) {
+        if(!value.matches("[A-Za-z][A-Za-z0-9_]*") && value.length() > 0) {
```

Comment:
Ugh.   I don't have an idea


== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty


On Wed, Jun 17, 2015 at 12:42 AM, KeithGalli <notifications@github.com>
wrote:

> In
> appinventor/appengine/src/com/google/appinventor/client/wizards/youngandroid/NewYoungAndroidProjectWizard.java
> <https://github.com/mit-cml/appinventor-sources/pull/402#discussion_r32594975>
> :
>
> > @@ -43,7 +47,32 @@ public NewYoungAndroidProjectWizard() {
> >      // Initialize the UI
> >      setStylePrimaryName("ode-DialogBox");
> >
> > -    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel());
> > +    projectNameTextBox = new LabeledTextBox(MESSAGES.projectNameLabel(), new Validator() {
> > +      @Override
> > +      public boolean validate(String value) {
> > +        if(!value.matches("[A-Za-z][A-Za-z0-9_]*") && value.length() > 0) {
>
> My problem is that GWT doesn't seem to support the java.util.regex
> libraries, so I can't use the methods described in that link. GWT uses a
> package called RegExp, but that doesn't have all the functionality of a
> pure Java implementation.
>
>
> http://www.gwtproject.org/javadoc/latest/com/google/gwt/regexp/shared/RegExp.html
>
> This package deals with regexes in the same way as javascript does.
> Unfortunately it seems like javascript doesn't have any built in libraries
> that are good for dealing with unicode characters so it is pretty difficult
> to match international letters. See the second response to this question.
>
>
> http://stackoverflow.com/questions/150033/regular-expression-to-match-non-english-characters
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mit-cml/appinventor-sources/pull/402/files#r32594975>.
>